### PR TITLE
connect-plug-ovn-certificates: gracefully handle /etc/ovn being a symlink

### PR DESCRIPTION
**Pull request by simondeziel**

### CODE REVIEW

File: snapcraft/hooks/connect-plug-ovn-certificates [hunk 1]
```diff
@@ -37,8 +37,10 @@ get_bool() {
 if [ -e "/etc/.lxd_generated" ]; then
   ovn_builtin=$(get_bool "$(snapctl get ovn.builtin)")
 
-  if test -d /etc/ovn ; then
```
Comment (simondeziel) [inline snapcraft/hooks/connect-plug-ovn-certificates:40]: `/etc/ovn` can be symlink pointing to `/var/lib/snapd/hostfs/etc/ovn`. Since `test -d`|`[ -d ... ]` follow symlinks, that `if` will succeed if the final directory `/var/lib/snapd/hostfs/etc/ovn` exists. In this case, is it really a good idea to do a `rm -rf` of it?

I think we should first check if it's a symlink `[ -L ... ]` (which does not follow links ;) ) and if not, only then check if it's a dir.

In other words, I'm asking if we should reverse the `if`/`elif` from that PR.

File: snapcraft/hooks/connect-plug-ovn-certificates [hunk 2]
```diff
@@ -37,14 +37,17 @@ get_bool() {
...
+    echo "=> Removing /etc/ovn symlink"
     rm -f /etc/ovn
   elif [ -d /etc/ovn ]; then
+    echo "=> Removing /etc/ovn directory"
     rm -rf /etc/ovn
   fi
 
   if ! [ "${ovn_builtin:-"false"}" = "true" ]; then
+    echo "=> Configuring MicroOVN Content Interface"
     mkdir -p /etc/ovn
```
Comment (roosterfish) [inline snapcraft/hooks/connect-plug-ovn-certificates:51]: Do we still need the `-p` here? I am wondering if it would hide a bug in the cleanup if we still have a `/etc/ovn` directory at this stage?

**Review by tomponline — APPROVED** [review]

File: snapcraft/hooks/connect-plug-ovn-certificates (hunk 2 above)
Comment (simondeziel) [reply snapcraft/hooks/connect-plug-ovn-certificates:51]: I think you are right, that `-p` is superfluous. However I'm on the fence on whether to drop it or not as ideally `daemon.start` should be doing the same, IMHO. I'm hesitant to touch `daemon.start` more :/

Comment (roosterfish) [reply snapcraft/hooks/connect-plug-ovn-certificates:51]: Ok fine for me, shouldn't be a blocker.

**Review by roosterfish — APPROVED** [review]

**Comment by tomponline** [comment]
@simondeziel can you let me know when this is in latest/edge and tested working ta

**Comment by simondeziel** [comment]
> @simondeziel can you let me know when this is in latest/edge and tested working ta

Looks like it passed the microcloud test suite this time!
